services/slam/fake/slam_test.go

```diff
@@ -170,3 +240,88 @@ func reverse[T any](slice []T) []T {
 	}
 	return slice
 }
+
+func verifyGetPointCloudMapStreamStateful(t *testing.T, mimeType string, slamSvc *SLAM, extra map[string]interface{}) {
+	testDataCount := maxDataCount
+	getMapPcdResults := []float64{}
+	positionResults := []spatialmath.Pose{}
+	getPositionResults := []spatialmath.Pose{}
+	getInternalStateResults := []int{}
+
+	// Call GetPointCloudMapStream twice for every testData artifact
+	for i := 0; i < testDataCount*2; i++ {
+		pInFrame, err := slamSvc.Position(context.Background(), slamSvc.Name, extra)
+		test.That(t, err, test.ShouldBeNil)
+		positionResults = append(positionResults, pInFrame.Pose())
+
+		p, _, err := slamSvc.GetPosition(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		getPositionResults = append(getPositionResults, p)
+
+		data, err := slamSvc.GetInternalState(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		getInternalStateResults = append(getInternalStateResults, len(data))
+
+		f, err := slamSvc.GetPointCloudMapStream(context.Background(), slamSvc.Name)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, f, test.ShouldNotBeNil)
+		pcd, err := helperConcatenateChunksToFull(f)
+		test.That(t, err, test.ShouldBeNil)
+		pc, err := pointcloud.ReadPCD(bytes.NewReader(pcd))
+		test.That(t, err, test.ShouldBeNil)
+		vObj, err := vision.NewObject(pc)
```

Comment:
[q] do we need to cast to a vObj, pointcloud metadata can be used instead to same affect

Reply:
changed